File: _includes/code/llms-txt/typescript/queries.ts
// llms.txt snippet: queries (nearText, bm25). Section "Python / TypeScript > Queries".
import weaviate, { vectors } from 'weaviate-client';

const client = await weaviate.connectToWeaviateCloud(process.env.WEAVIATE_URL!, {
  authCredentials: new weaviate.ApiKey(process.env.WEAVIATE_API_KEY!),
});
await client.collections.delete('Movie__QueriesTs');
await client.collections.create({
  name: 'Movie__QueriesTs',
  vectorizers: vectors.text2VecWeaviate(),
});
const col = client.collections.use('Movie__QueriesTs');
await col.data.insertMany([
  { title: 'The animals of the savannah' },
  { title: 'A bowl of ramen and other street food' },
]);

// START llms_queries
// Vector search
const vectorRes = await col.query.nearText('animals in movies', { limit: 3, returnMetadata: ['distance'] });
// Keyword search
const keywordRes = await col.query.bm25('food', { limit: 3, returnMetadata: ['score'] });
// END llms_queries

if (keywordRes.objects.length < 1) throw new Error('no bm25 results');
await client.collections.delete('Movie__QueriesTs');
await client.close();

File: _includes/code/llms-txt/python/queries.py
"""llms.txt snippet: queries (near_text, bm25). Section "Python / TypeScript > Queries"."""
import os
import weaviate
from weaviate.classes.init import Auth
from weaviate.classes.config import Configure

client = weaviate.connect_to_weaviate_cloud(
    cluster_url=os.environ["WEAVIATE_URL"],
    auth_credentials=Auth.api_key(os.environ["WEAVIATE_API_KEY"]),
)

client.collections.delete("Movie__QueriesPy")
client.collections.create(
    "Movie__QueriesPy",
    vector_config=Configure.Vectors.text2vec_weaviate(),
)
col = client.collections.use("Movie__QueriesPy")
col.data.insert_many([
    {"title": "The animals of the savannah"},
    {"title": "A bowl of ramen and other street food"},
])

# START llms_queries
from weaviate.classes.query import MetadataQuery
# Vector search
res = col.query.near_text("animals in movies", limit=3, return_metadata=MetadataQuery(distance=True))
# Keyword search
res = col.query.bm25("food", limit=3, return_metadata=MetadataQuery(score=True))
# END llms_queries

assert len(res.objects) >= 1
client.collections.delete("Movie__QueriesPy")
client.close()
